### ISSUE TITLE
AP can't keep a certain speed, but it throttles up and down violently

### ISSUE THREAD

**Issue by franchiceddhu** [flybywiresim/a32nx#4436]
**Mod Version**

```
{
  "built": "2021-04-14T11:41:57+00:00",
  "ref": "autopilot",
  "sha": "1fa16604bfbedef2a17e856903ee1c40e4784c28",
  "actor": "lukecologne",
  "event_name": "manual"
}
```

**Describe the bug**

The AP isn't able to keep managed nor selected speed with A/THR but instead it "bounces" throttling up and down.

**To Reproduce**

1. I set a flight to VGHS airport.
2. After descent, when plane has to be leveled at a certain altitude, the plane starts its "bouncing".
3. Same problem keeps with APPR phase on and on G/S (but not that harshly).
4. I've been able to reproduce it multiple times.

**Expected behavior**

I expect the plane to keep its flight leveled and at constant speed.

**Actual behavior**

The throttle goes to near-idle to 80-90% sometimes, even quite violently. The first time it occurred, the speed got so low it eventually stalled. I had to adjust throttle manually till airport.

**References**

[https://www.youtube.com/watch?v=OqiW6iOSC5U](url) (just loaded a flight from a random location nearby VGHS airport to reproduce the issue. I didn't do a proper checklist and tried x-feed, but as far as I know, this issue was exactly the same as my proper flight with regular checklist, when I firstly encountered it).

**Additional context**

Was this working before/when did the issue start occurring?
I never had this problem before 1.15.7.0. These are my first flights with the new version.

Is this a problem in the vanilla unmodded game?
ND

**Comment by Watsi01** [comment]
I had kind of the same issue once, approaching an airport with very hot outside temp. Do you remember what the temp in Dhaka was?

**Comment by Watsi01** [comment]
Also @tyler58546

**Comment by franchiceddhu** [comment]
Really interesting. It was indeed hot. I use live weather and I landed at 12:00 Zulu, when the sun was about to set. 34 degrees C, if I'm not wrong (now, at 16:00Z, it's 31° C and I just tried to land with the same issue).

**Comment by MisterChocker** [comment]
Well, on the lower ecam you have also temperature sensor and the TAT 34°C.

**Comment by aguther** [comment]
It would be good to get the FDR file of that flight to further investigate.

**Comment by franchiceddhu** [comment]
> It would be good to get the FDR file of that flight to further investigate.

Is this an actual feature that I'm totally unaware of, or just a wish? It would be a developer's dream.

**Comment by Watsi01** [comment]
@franchiceddhu 
Depending on your version of MSFS (Steam or MSStore) there's a MSFS folder inside your Windows user folder, (in Appdata/Roaming or local or somewhere there). This MSFS folder also contains a "Packages" folder. There should be a flybywire-aircraft-a320-neo/work folder. Inside that, you find .fdr files. Check if you still have files dated from the mentioned flight you made. These files would be interesting.

**Comment by franchiceddhu** [comment]
> @franchiceddhu
> Depending on your version of MSFS (Steam or MSStore) there's a MSFS folder inside your Windows user folder, (in Appdata/Roaming or local or somewhere there). This MSFS folder also contains a "Packages" folder. There should be a flybywire-aircraft-a320-neo/work folder. Inside that, you find .fdr files. Check if you still have files dated from the mentioned flight you made. These files would be interesting.

Didn't knew this was a thing! I found three .fdr files that probably refer to my trials (I've made 2 or 3 more, but I'm not sure if they are an actual flight).

Had to put in a dropbox since they are really big.
[AP throttle issue](https://www.dropbox.com/sh/l9lh8dn74lkxn9b/AAAGSwS1NOVD_vjZolOYQUwxa?dl=0)

2021-04-14-07-34-01.fdr refers to my flight when I firstly discover the issue and the plane stalled. You may see me trying to disconnect AP multiple times. It was a somehow cursed flight, since the departure airport (ZUGY) had no procedures and an outdated runway, despite my Navigraph addon.

2021-04-14-14-00-53.fdr should be the one from the video I sent. Departed from a random location about 100 NM from VGHS airport (the same as the flight before), used a different ILS VIA this time.

2021-04-14-14-54-43.fdr should be another try, but I'm not sure about it and what I actually did. If this was again near VGHS, it's another try - if not, please ignore it.

**Comment by Watsi01** [comment]
Thanks :-)

**Comment by aguther** [comment]
For unknown reasons I was unable to read the FDR file. I've now added an additional interface version to those FDR files so that we know better in future.

I tried to reproduce and it seems something related to warm weather:
https://youtu.be/MjXXAS3mE7E
![image](https://user-images.githubusercontent.com/6629914/114927087-6204cd00-9e31-11eb-9ea6-1dbeb6b6bfd3.png)

**Comment by franchiceddhu** [comment]
> I was unable to read the FDR file

Sorry to hear that, were all of them inaccessible? I've just included a .zip file in my Dropbox folder, in the remote case the upload would have messed up the files.

Thanks for your commitment. One further piece of info just came up to my mind: on April 12th I was flying from Abu Dhabi to Jeddah with live weather, landed at 11:30Z, and the temperature was about 37° C at departure and 30° C at arrival, yet my AP worked fine as usual. I was using the same experimental version as my flight to VGHS; the only differences were the 1.15.7.0 version and a custom livery, both for the VGHS flights.

**Comment by Taz5150** [comment]
Found the issue and its been corrected as we speak. 

Thank you for the feedback!

**Comment by aguther** [comment]
I reverted a change, so build is running. After that is finished, please update to latest experimental version using the installer and try again. Please close the issue when it's solved.

**Comment by franchiceddhu** [comment]
@aguther @Taz5150 

Thanks again. 
Since weather conditions have changed in VGHS, I tried a custom flight with custom weather near the same airport, before updating my experimental version, and the issue could be reproduced at temperatures higher than 30° C.

I then updated using the installer. This time, using the same presets, the issue was not there! But, when increasing the temperatures even further, in the 40-50° C range, the issue showed up once again. 

I got another video about the issue showing after some temperatures thresholds:

[AP issue with extremely high temperatures](https://www.youtube.com/watch?v=d5JwBAPN1rE) (video is still being elaborated in HD as I'm writing)

Now, I was unsure if the issue was due to the fact I had not specified any PERF Dest data, so I tried with an arbitrary QNH and TEMP, but the issue apparently was still there.

For me, your fix has been a huge improvement since it's hard to see such hot temperatures using live weather, but as a total noob, I'm unsure if this is an actual engines behaviour or a bug. So I'm quite unsure if I should mark this issue as "closed": let me know!

**Comment by sidnov** [comment]
This is exactly what I was experiencing with the thrust going haywire when flying around the Indian FIRs. Similar weather conditions :D

**Comment by Watsi01** [comment]
@franchiceddhu Yeah, there still seems to be a problem, depending on the outside temperature. The question is, where to set the treshold for 'realistic' temps and say: if the fadec 'brakes' above that, it doesn't matter.  I think we should keep that open until a decision is made by @aguther and @Taz5150.

**Comment by leonpo** [comment]
same happened for me with 35C temp.

**Comment by Taz5150** [comment]
Will re-test and be back with a solution. There are other parameters that could affect engine spool oscillations. I suppose everyone is working with the latest experimental after Andreas reverted changes?

Thanks.

**Comment by franchiceddhu** [comment]
I do, yes..

**Comment by CptChaos0815** [comment]
Another example of this problem recoreded at Medina OEMA at 35°C OAT.

{
  "built": "2021-04-19T18:55:39+00:00",
  "ref": "experimental",
  "sha": "c0b494cd0fc448ade05bc1a394fcbdaaae3e5de8",
  "actor": "aguther",
  "event_name": "manual"
}

FDR file:

[2021-04-21-08-29-10.zip](https://github.com/flybywiresim/a32nx/files/6349337/2021-04-21-08-29-10.zip)

Video:

**Comment by MisterChocker** [comment]
Why haven't you put the THR LVR into CLB as requested by the PFD?

**Comment by Taz5150** [comment]
Engine oscillation has been completely solved with PR #4553 ... testing it as we speak. If things go well, this issue should be solved.
